src/Sentry/DynamicSamplingContext.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Sentry.Internal.Extensions;
+
+namespace Sentry
+{
+    /// <summary>
+    /// Provides the Dynamic Sampling Context for a transaction.
+    /// </summary>
+    /// <seealso href="https://develop.sentry.dev/sdk/performance/dynamic-sampling-context"/>
+    internal class DynamicSamplingContext
+    {
+        public IReadOnlyDictionary<string, string> Items { get; }
+
+        public bool IsEmpty => Items.Count == 0;
+
+        private DynamicSamplingContext(IReadOnlyDictionary<string, string> items) => Items = items;
+
+        /// <summary>
+        /// Gets an empty <see cref="DynamicSamplingContext"/> that can be used to "freeze" the DSC on a transaction.
+        /// </summary>
+        public static readonly DynamicSamplingContext Empty = new(new Dictionary<string, string>().AsReadOnly());
```

Comment:
remove the `.AsReadOnly()`

Reply:
Why?  It ensures it can't ever be mutated, and also there's only one instance so there's not much overhead.